sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -390,8 +390,20 @@ RT::PiProgram ProgramManager::getBuiltPIProgram(OSModuleHandle M,
         !SYCLConfig<SYCL_DEVICELIB_NO_FALLBACK>::get())
       DeviceLibReqMask = getDeviceLibReqMask(Img);
 
+    std::string CompileOpts = Img.getCompileOptions();
+    if (Img.getSyclIsEsimdImage().isAvailable()) {
+      bool IsEsimdImage =
+          pi::DeviceBinaryProperty(*(Img.getSyclIsEsimdImage().begin()))
+              .asUint32();
+      if (IsEsimdImage) {
+        if (!CompileOpts.empty())
+          CompileOpts += " ";
+        CompileOpts += "-vc-codegen";
```

Comment:
Could you please tell what happens if a context is bound to non-Gen device?

Reply:
I think nothing will happen, we will not add `-vc-codegen` option. The new `IsEsimdImage` property will be added only for images that have ESIMD kernels inside. See this patch: https://github.com/intel/llvm/pull/3073/commits/557b564049afbb8bb4d7a6bf22ff15fb0559b3e3

Reply:
An assert would be helpful then, I think. It would check that the context is intel GPU, as other BEs will crash/ignore this option.